### PR TITLE
Expose BundleOptions and remove __namedParameters

### PR DESCRIPTION
**Pull request by lorensr**
- chore(docs): Prevent __namedParameters params from being generated
- chore(docs): Expose BundleOptions

Current docs:

image

Proposed:

image

### CODE REVIEW

File: packages/worker/src/logger.ts [hunk 1]
```diff
@@ -43,7 +45,8 @@ function defaultLogFunction({ level, timestampNanos, message, meta }: LogEntry):
 /**
  * Takes a `[seconds, nanos]` tuple as returned from getTimeOfDay and turns it into bigint.
  */
-export function timeOfDayToBigint([seconds, nanos]: [number, number]): bigint {
+export function timeOfDayToBigint(timeOfDay: [number, number]): bigint {
```
Comment (bergundy) [inline packages/worker/src/logger.ts:48]: I don't think we expose this to users, we can keep it they way it was

File: packages/worker/src/logger.ts [hunk 2]
```diff
@@ -31,7 +31,9 @@ const severities: LogLevel[] = ['TRACE', 'DEBUG', 'INFO', 'WARN', 'ERROR'];
 /**
  * Log messages using `console.error` and basic formatting
  */
-function defaultLogFunction({ level, timestampNanos, message, meta }: LogEntry): void {
+function defaultLogFunction(entry: LogEntry): void {
```
Comment (bergundy) [inline packages/worker/src/logger.ts:34]: We don't expose this to users, we can keep it they way it was

**Review by bergundy — COMMENTED** [review]
Thanks, looks like a typedoc limitation, right?

**Comment by lorensr** [comment]
If we upgraded, we could alternatively use `@param` comments. 

image

It would be more verbose in cases when we already have a type like `BundleOptions`.

Trying latest (`typedoc@0.22.10`) didn't immediately work for me, and after brief Googling, I didn't dig:

image

File: packages/worker/src/logger.ts (hunk 1 above)
Comment (lorensr) [reply packages/worker/src/logger.ts:48]: I only updated functions that show up in search:

https://typescript.temporal.io/search?q=__namedPara

image

**Review by bergundy — APPROVED** [review]
Thanks @lorensr, I'l try to get this to pass CI after I merge https://github.com/temporalio/sdk-typescript/pull/414

**Pull request merged**